### PR TITLE
Added github workflows and removed travis

### PR DESCRIPTION
**Pull request by nils-braun**
Solves #74 (except the bumpversion, which can not be solved as we are not using git flow).

So far, whenever there is a new tag, there is in principle also a pipy publish via Travis. However, Travis is a bit flaky sometimes and GitHub actions offer more possibilities anyhow, so let's move to GitHub actions :-)

This PR removes the Travis integration and adds two GitHub actions: one for publishing (whenever a new GitHub release is created, which typically should be done on git tags) and one for the usual tests of PRs. I also moved to codecov in this PR (just because it is nicer). 

This can be expanded to also include things like stylechecks (pycodestyle, autopep8, black), isort, flake8 etc - but so far letÄs start with the basics.

### CODE REVIEW

**Comment by nils-braun** [comment]
@meliache Feel free to either merge it if you like to comment on it. That is of course only my own proposal

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
```
Comment (meliache) [inline .github/workflows/test.yml:19]: Just fyi, as you can't know, but the current basf2 externals still use python 3.6.10, so the code should work with that, but I don't think it's an issue, I just [checked](https://docs.python.org/3.7/whatsnew/3.7.html)  there is no new syntax in 3.7 (walrus only appears in 3.8)

**Comment by meliache** [comment]
Thanks, looks good to me :slightly_smiling_face:. And if you agree with my one comment above that testing with 3.7 is probably no problem, then I'll happily merge.

File: .github/workflows/test.yml (hunk 1 above)
Comment (nils-braun) [reply .github/workflows/test.yml:19]: We can also check with both 3.6 and 3.7.
Let's do this in another PR (at a later point in time)

**Pull request merged**